src/UnleashBuilder.php

```diff
@@ -376,7 +375,82 @@ public function withMetricsBucketSerializer(?MetricsBucketSerializer $metricsBuc
         return $this->with('metricsBucketSerializer', $metricsBucketSerializer);
     }
 
+    public function buildRepository(): UnleashRepository
+    {
+        $dependencyContainer = $this->initializeContainerWithConfiguration();
+
+        assert($dependencyContainer->getConfiguration() !== null);
+
+        return new DefaultUnleashRepository($dependencyContainer->getHttpClient(), $dependencyContainer->getRequestFactory(), $dependencyContainer->getConfiguration());
+    }
+
     public function build(): Unleash
+    {
+        $dependencyContainer = $this->initializeContainerWithConfiguration();
+        assert($dependencyContainer->getConfiguration() !== null);
+
+        $repository = new DefaultUnleashRepository($dependencyContainer->getHttpClient(), $dependencyContainer->getRequestFactory(), $dependencyContainer->getConfiguration());
```

Comment:
This should use the `buildRepository()` method to not duplicate logic. Though I don't like the idea of building the container twice like that, so perhaps a private method, that will build the repository and accepts nullable container as a parameter?

If you can think of other solutions, feel free to implement them.

Reply:
I extracted it to a private method `createRepository`.

Reply:
Yes, looks great, thanks!